ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx

```diff
@@ -27,25 +30,44 @@ const TypedSignInfo: React.FC = () => {
     return null;
   }
 
-  const { domain = {} } = JSON.parse(
-    currentConfirmation.msgParams.data as string,
-  );
+  const {
+    domain,
+    domain: { verifyingContract },
+    primaryType,
+  } = parseTypedDataMessage(currentConfirmation.msgParams.data as string);
```

Comment:
Leaving the same comment here https://github.com/MetaMask/metamask-extension/pull/24862#discussion_r1625520301